libwayshot/src/lib.rs

```diff
@@ -491,7 +497,10 @@ impl WayshotConnection {
                 Ok(())
             })?;
         }
-        Ok(())
+        let region = callback()?;
+        layer_shell.destroy();
+        event_queue.blocking_dispatch(&mut state)?;
```

Comment:
If the callback fails, we'll not destroy the layer shell. Can you make it so regardless of the return value, we'll still destroy the layer shell and try to send it on the event queue?